src/model_repository_manager.cc

```diff
@@ -768,24 +822,63 @@ ModelRepositoryManager::LoadUnloadModels(
   }
   std::set<ModelIdentifier> deleted_dependents;
 
-  // Update dependency graph and load
-  UpdateDependencyGraph(
-      added, deleted, modified,
+  // Make a copy of the dependency graph for rollback if there is a collision
+  // between the models to be loaded/unloaded by this thread with another
+  // thread actively loading/unloading models. The global map referenced by the
+  // dependency graph is managed by this object, which is the responsibility of
+  // this object to make a copy.
+  std::unordered_map<std::string, std::set<ModelIdentifier>> global_map(
+      global_map_);
+  DependencyGraph dependency_graph(dependency_graph_, &global_map);
+
+  // Update the dependency graph independently for the possibility of rollback.
+  auto affected_models = UpdateDependencyGraph(
```

Comment:
I think this function and maybe other doesn't really access any member variables after this change?

Reply:
Yes.

Reply:
Update commit: [Refactor parallel load logic](https://github.com/triton-inference-server/core/pull/165/commits/0fa4e296b613806d0bd7de77775f09e51d866d21)